src/libraries/System.Net.Security/src/System/Net/Security/SslStream.Implementation.cs

```diff
@@ -249,35 +240,23 @@ private int CheckOldKeyDecryptedData(Memory<byte> buffer)
         // This method assumes that a SSPI context is already in a good shape.
         // For example it is either a fresh context or already authenticated context that needs renegotiation.
         //
-        private void ProcessAuthentication(LazyAsyncResult lazyResult, CancellationToken cancellationToken)
+        private void ProcessAuthentication()
         {
             if (Interlocked.Exchange(ref _nestedAuth, 1) == 1)
             {
-                throw new InvalidOperationException(SR.Format(SR.net_io_invalidnestedcall, lazyResult == null ? "BeginAuthenticate" : "Authenticate", "authenticate"));
+                throw new InvalidOperationException(SR.Format(SR.net_io_invalidnestedcall, "Authenticate", "authenticate"));
             }
```

Comment:
I'm confused about some of the changes in this PR.  Previously ProcessAuthentication was called for both sync and async entrypoints, but now it's only used by sync.  For the async path, what handles changing _nestedAuth, calling ThrowIfException, etc.?

Reply:
I was not clear to me if we really needed it. I may be wrong but I assume the exceptions now simply pass up - using await in Async case. The _nestedAuth only guards agains incorrect API use, right? If so is there some way how to avoid use of Interlocked.Exchange()? 
And I realized I missed call to NetEventSource.Log() by skipping ProcessAuthentication() and calling ForceAuthentication() directly. I'll add that also to path for renegotiation.

Reply:
> The _nestedAuth only guards agains incorrect API use, right?

Yes

> If so is there some way how to avoid use of Interlocked.Exchange()?

Not while still guarding usage in a thread-safe manner.

Reply:
I added ProcessAuthentication() back as entry point for all cases.